pkg/db/mock/executor.go

```diff
@@ -0,0 +1,105 @@
+package mock
+
+import (
+	"time"
+
+	"github.com/Jguer/yay/v11/pkg/db"
+
+	"github.com/Jguer/go-alpm/v2"
+)
+
+type (
+	IPackage = alpm.IPackage
+	Depend   = alpm.Depend
+	Upgrade  = db.Upgrade
+)
+
+type DBExecutor struct{}
```

Comment:
To avoid implementing all the methods you could have added the interface to the struct but now that it's done :shrug:

Reply:
Like what I did down here, you mean?

https://github.com/Jguer/yay/pull/1625/files#diff-f393936084bd48dbb2beec1877beb76e5301b7b896abf272a5298f619b1a9f6aR190-R201

Reply:
Yeah, sometimes it leads to weird bugs though